### PR TITLE
Use nocookie domain for YT videos

### PR DESCRIPTION
**Pull request by regiskuckaertz**
## What does this change?

Switches YT videos to the `youtube-nocookie.com` domain so that a cookie is only placed on the user's browser when they hit the play button.

## What is the value of this and can you measure success?

This is for GDPR-compliance reasons.

### Does this affect other platforms?

- [ ] AMP: perhaps, but I haven't been able to run `amp-validate` locally

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [x] Yes

_All_ YT videos are affected.

### Does this change break ad-free?

- [x] No

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: onward/app/controllers/CardController.scala [hunk 1]
```diff
@@ -37,6 +37,7 @@ class CardController(
       "www.dailymotion.com",
       "youtube.com",
       "www.youtube.com",
+      "www.youtube-nocookie.com",
```
Comment (regiskuckaertz) [inline onward/app/controllers/CardController.scala]: I don't know if that's necessary, but thought it wouldn't hurt anyway. Can remove if not needed.

File: static/src/javascripts/projects/common/modules/atoms/youtube.js [hunk 2]
```diff
@@ -154,66 +145,21 @@ const handlePlay = (atomId: string, player: AtomPlayer): void => {
...
-    /**
-     * Get the youtube video id from the video currently playing.
-     * We want to compare with the youtube ID in memory. If they differ
-     * a related video has begun playing, so we need to get the atom ID
-     * for tracking.
-     */
-    const latestYoutubeId = getYoutubeIdFromUrl(youtubePlayer.getVideoUrl());
+    const { youtubePlayer, atomId } = players[uniqueAtomId];
 
-    if (youtubeId !== latestYoutubeId) {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/atoms/youtube.js]: 🤔 i remember wondering how that would be possible? @akash1810 ?

Comment (regiskuckaertz) [reply static/src/javascripts/projects/common/modules/atoms/youtube.js]: After discussing with Akash, it seems that whole logic was used for an AB test and shouldn't be necessary anymore. Can you confirm, @GHaberis or @nicl or @philmcmahon ? 😄

Comment (GHaberis) [reply static/src/javascripts/projects/common/modules/atoms/youtube.js]: @regiskuckaertz this is still necessary, `youtubeId` and `latestYoutubeId` can become unequal once a user starts playing a related video from the videos presented to them using youtube's native "related videos" feature.

**Comment by regiskuckaertz** [comment]
[Frontend - Use the nocookie domain for youtube videos](https://trello.com/c/fkVww9KK/773-frontend-use-the-nocookie-domain-for-youtube-videos)

File: static/src/javascripts/projects/common/modules/atoms/youtube.js (hunk 2 above)
Comment (akash1810) [reply static/src/javascripts/projects/common/modules/atoms/youtube.js]: From memory this was used during the A/B test. That is, success of the test to move to the standard yt endslate was to track onward journeys using video starts as the metric. Now that the test is complete and the yt endslate won, I don't think we need to continue this tracking?

Comment (GHaberis) [reply static/src/javascripts/projects/common/modules/atoms/youtube.js]: >  I don't think we need to continue this tracking?

We'd need to confirm this with Matt Ho. The tracking this enables doesn't just cover video starts, but also the additional milestones (25, 50, 75% played etc), if we stop tracking this then video plays triggered from youtube's endslate would not appear in our own tracking data.

Comment (regiskuckaertz) [reply static/src/javascripts/projects/common/modules/atoms/youtube.js]: I feel like this is a discussion for another PR. I'll speak with Matt, thanks both for your help 👍

**Comment by regiskuckaertz** [comment]
Can someone confirm if this is used in AMP pages? If not, I've tested events were being sent out as usual.

**Comment by regiskuckaertz** [comment]
Can I get a review on this?

File: common/app/views/fragments/atoms/youtube.scala.html [hunk 3]
```diff
@@ -58,7 +58,7 @@
 
         @for(asset <- activeAsset if playable && !expired) {
 
-            <div id="youtube-@asset.id" data-asset-id="@asset.id" class="youtube-media-atom__iframe"></div>
+            <iframe id="youtube-@asset.id" data-asset-id="@asset.id" class="youtube-media-atom__iframe" src="https://www.youtube-nocookie.com/embed/@{asset.id}?enablejsapi=1"></iframe>
```
Comment (GHaberis) [inline common/app/views/fragments/atoms/youtube.scala.html]: @regiskuckaertz I have a question regarding swapping out the placeholder for an `iframe`. Currently when we initialise the `new window.YT.Player` (here:  https://github.com/guardian/frontend/blob/fd4256415a69a9911f62ae6fc2761979763df9fd/static/src/javascripts/projects/common/modules/atoms/youtube-player.js#L108) we pass `eltId ` which is the placeholder `div`'s ID. I believe the Youtube script then inserts an iframe into this placeholder, or replaces it with an iframe. 

For performance reasons on fronts we now wait to set up the player until the user clicks the placeholder, this means we wait to call `new window.YT.Player` which in turn inserts the `iframe`. The reason this was done was so we didn't preload multiple `iframe`s in the video carousel. the following PR made that change: https://github.com/guardian/frontend/pull/20286

Now we have the iframe there by default will we be loading all these `iframe`s and there assets to begin?

Comment (regiskuckaertz) [reply common/app/views/fragments/atoms/youtube.scala.html]: Yes 😒

Comment (mchv) [reply common/app/views/fragments/atoms/youtube.scala.html]: @regiskuckaertz You can simply configure the `host` using `window['YTConfig'].host = "www.youtube-nocookie.com"` creating `YTConfig` if it does not exist.

Comment (mchv) [reply common/app/views/fragments/atoms/youtube.scala.html]: This way you can keep the `placeholder` and only update `youtube-player.js`

Comment (regiskuckaertz) [reply common/app/views/fragments/atoms/youtube.scala.html]: huhoo, some hidden feature. It actually didn't work but it turns out you can set the `host` setting as a `YT.Player` setting, also undocumented but it worked 🎉 Thanks Mariot!

**Comment by regiskuckaertz** [comment]
Happy with this @GHaberis ?

**Review by shtukas — APPROVED** [review]

**Comment by regiskuckaertz** [comment]
Merci 🙌

**Pull request merged**